storage/src/test/java/org/apache/kafka/tiered/storage/integration/ReassignReplicaExpandTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.tiered.storage.integration;
+
+import org.apache.kafka.tiered.storage.TieredStorageTestBuilder;
+import org.apache.kafka.tiered.storage.TieredStorageTestHarness;
+import org.apache.kafka.tiered.storage.specs.KeyValueSpec;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+
+public class ReassignReplicaExpandTest extends TieredStorageTestHarness {
+
+    protected final Integer broker0 = 0;
+    protected final Integer broker1 = 1;
+
+    /**
+     * Cluster of two brokers
+     * @return number of brokers in the cluster
+     */
+    @Override
+    public int brokerCount() {
+        return 2;
+    }
+
+    /**
+     * Number of partitions in the '__remote_log_metadata' topic
+     * @return number of partitions in the '__remote_log_metadata' topic
+     */
+    @Override
+    public int numRemoteLogMetadataPartitions() {
+        return 2;
+    }
+
+    @Override
+    protected void writeTestSpecifications(TieredStorageTestBuilder builder) {
+        final String topicA = "topicA";
+        final String topicB = "topicB";
+        final Integer p0 = 0;
+        final Integer partitionCount = 5;
+        final Integer replicationFactor = 2;
+        final Integer maxBatchCountPerSegment = 1;
+        final Map<Integer, List<Integer>> replicaAssignment = null;
+        final boolean enableRemoteLogStorage = true;
+        final List<Integer> metadataPartitions = new ArrayList<>();
+        for (int i = 0; i < numRemoteLogMetadataPartitions(); i++) {
+            metadataPartitions.add(i);
+        }
+
+        builder
+                // create topicA with 5 partitions and 2 RF
+                .createTopic(topicA, partitionCount, replicationFactor, maxBatchCountPerSegment,
```

Comment:
What is the reason for creating topicA? No messages are sent to that topic and the respective checks. But I see a check whether metadata partitions are assigned or not for this topic.

Reply:
yes correct.